### DISCUSSION TITLE
Integrazione sensore PIR con Local Tuya

### DISCUSSION THREAD

**Discussion by braudoo** [indomus/forum#123] [HUB: Home Assistant]
Ciao a tutti, sto cercando in tutti modi ad integrare un sensore PIR wifi, ma senza successo.
La mia configurazione è HA Core installato come app su Raspbian
Di seguito gli step che ho seguito:
- installato Local Tuya con HACS
- seguito guida sul repositary di Local Tuya per identificare devide KEy e ID
- aggiunto alla configuration.yaml il device + entità:

```
localtuya:
  - host: 192.168.1.x
    device_id: 8242418584f3ebfxxxxx
    local_key: d0a2a84d72f7xxxxxx
    friendly_name: Tuya Device
    protocol_version: "3.3"
    entities:
      - platform: binary_sensor
         friendly_name: PIR Status
         id: 1
         device_class: motion
         state_on: "true" # Optional
         state_off: "false" # Optional
```

-riavviato HA

Il device non è disponibile e questo è il LOG che mi restituisce:

```
Logger: custom_components.localtuya.common
Source: custom_components/localtuya/pytuya/init.py:646
Integration: LocalTuya integration (documentation, issues)
First occurred: 23:12:53 (2 occurrences)
Last logged: 23:13:52

[824...8fb] Connect to 192.168.1.3 failed
Traceback (most recent call last):
File "/home/homeassistant/.homeassistant/custom_components/localtuya/common.py", line 144, in _make_connection self,
File "/home/homeassistant/.homeassistant/custom_components/localtuya/pytuya/init.py", line 646, in connect port,
File "/usr/lib/python3.7/asyncio/base_events.py", line 959, in create_connection raise exceptions[0]
File "/usr/lib/python3.7/asyncio/base_events.py", line 946, in create_connection await self.sock_connect(sock, address)
File "/usr/lib/python3.7/asyncio/selector_events.py", line 464, in sock_connect return await fut
File "/usr/lib/python3.7/asyncio/selector_events.py", line 494, in _sock_connect_cb raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.3', 6668)
```

E' strano perchè sembra che non riesce a collegare l'ID all'host.

Qualcuno ha avuto problemi analoghi?

Grazie.

**Comment by snakuzzo** [comment]
Ciao, ti chiedo cortesemente di modificare il post utilizzando il markdown per tutto quello che riguarda codice e log, altrimenti sono costretto a rimuovere la discussione.
Grazie.

**Comment by braudoo** [comment]
Ok, messo.
avresti qualche lume in merito?

Grazie.

**Reply by snakuzzo** [reply]
beh...c'è poco da suggerire...la connessione è stata rifiutata dall'host `192.168.1.3`.
Scusa eh...ma quell' `host: 192.168.1.x` ??? Non dovresti mettere l'ip ? Che roba è quella `x` ?

**Comment by braudoo** [comment]
L'avevo usato per mascherare IP ma poi ho visto adesso che l'ho riportato nel LOG.
Ad ogni modo nel configuration.yaml ho messo host corretto (senza x :-D).
Ma alla base c'è qualche passaggio sbagliato?
Ho questo sensore PIR wifi e volevo integrarlo come dispotivo local push e ho visto questa come strada.
Se hai alternative da propormi, ben volentieri.
Grazie

**Reply by snakuzzo** [reply]
piccola nota...non ha alcun senso mascherare un IP di classe privata :D

Quanto al tuo problema...devi leggere la documentazione del custom component...se non funziona apri una issue sul progetto github del custom.
